discovery-infra/junit_log_parser.py

```diff
@@ -58,10 +59,11 @@ def get_failure_cases(log_file_name: Path, suite_name: str) -> List[TestCase]:
             fail_cases.append(test_case)
 
             # test is flaky
-            if entry.level != "fatal":
+            if entry.level != "fatal" and entry.func not in test_case_keys:
```

Comment:
let's do something else
define ``LogEntry`` as follows:
```
@dataclass
class LogEntry:
    time: str = None
    level: str = None
    msg: str = None
    func: str = dataclasses.field(hash=True)
    file: str = None
```
then you can have just ``fail_cases = set()``, adding items to it without fearing duplications

Reply:
I don't want that fail cases will be set, I just want to add the success case once (because that it is only for marking the test as flaky).
Note that the same function can fail multiple times and each failure is test case.

Reply:
Also, I thin for using set I have to also override `__hash__`

Reply:
``dataclasses.field(hash=True)`` automatically implements ``__hash__``
let me take another look at the code, I think we can still make it simpler

Reply:
Sure .. simpler is better  :)

```python
import dataclasses

@dataclasses.dataclass
class A:
    a: str = dataclasses.field(hash=True)
    b: str
    c: str

a = A("1", "2", "3")
b = A("5", "2", "3")
c = A("5", "2", "3")

print({a, b, c})
>>>>> TypeError: unhashable type: 'A'
```

Reply:
ah, it needs ``@dataclasses.dataclass(frozen=True)`` apparently